src/sage/quadratic_forms/binary_qf.py

```diff
@@ -1625,6 +1660,13 @@ def solve_integer(self, n):
             Q = self.matrix_action_right(M)
             assert not Q._c
 
+            if not Q._b:
+                # at this point, Q = a*x^2
+                if Q._a.divides(n) and (n // Q._a).is_square():
+                    x = (n // Q._a).isqrt()
+                    return tuple(row[0] * x for row in M.rows())
```

Comment:
```suggestion
                    return tuple([row[0] * x for row in M.rows()])
```
This (surprisingly) is faster. I am guessing you also checked that `tuple(M * x)` is slower?

Reply:
Done, and yes, by a factor of about 10.